### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by asm0dey** [koalaman/shellcheck#2087]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2004
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

fields=(background cursorColor)
colors=("#FFFAFA" "#CDC9C9" "#EE3B3B")
size=${#colors[@]}

for f in "${fields[@]}"; do
    index=$(($RANDOM % $size))
    color="${colors[$index]}"
    sed -i 's|\("'"$f"'"\s*:\s*"\)#.\+"|\1'"$color"'"|' myfile.json
done
```

#### Here's what shellcheck currently says:
```
Line 8	SC2004: $/${} is unnecessary on arithmetic variables.
Line 8	SC2004: $/${} is unnecessary on arithmetic variables.
```
#### Here's what I wanted or expected to see:

These look like false positives, they should be described better or be absent

**Comment by brother** [comment]
Looks correct to me really.

```
brother ~$ size=4
brother ~$ echo $(( $RANDOM % $size ))
3
brother ~$ echo $(( RANDOM % $size ))
0
brother ~$ echo $(( $RANDOM % size ))
0
brother ~$ echo $(( RANDOM % size ))
1
brother ~$ echo $(( RANDOM % size ))
3
```

**Comment by brother** [comment]
wiki page about warning. https://github.com/koalaman/shellcheck/wiki/SC2004

**Comment by koalaman** [comment]
The suggestion is correct as noted. What would you suggest as a better description?

**Comment by asm0dey** [comment]
Oh, sorry, I read warning incorrectly — I thought that it sees curly braces where they do not exist
